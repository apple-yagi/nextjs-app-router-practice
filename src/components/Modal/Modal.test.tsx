import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it } from "vitest";

import { Modal } from ".";

const ModalTest = () => (
  <Modal
    trigger={<button>Open Modal</button>}
    title="Modal Title"
    content="Modal Description"
  />
);

describe("Modal", () => {
  it("Modalが正しく表示されること", async () => {
    const user = userEvent.setup();
    const { getByRole } = render(<ModalTest />);

    await user.click(getByRole("button", { name: "Open Modal" }));

    expect(getByRole("dialog")).toBeInTheDocument();
    expect(getByRole("heading", { name: "Modal Title" })).toBeInTheDocument();
    expect(getByRole("dialog")).toMatchSnapshot();
  });

  it("CloseIconを押すとModalが閉じること", async () => {
    const user = userEvent.setup();

    const { getByRole, queryByRole } = render(<ModalTest />);

    await user.click(getByRole("button", { name: "Open Modal" }));
    expect(getByRole("dialog")).toBeInTheDocument();

    await user.click(getByRole("button", { name: "閉じる" }));
    expect(queryByRole("dialog")).not.toBeInTheDocument();
  });
});
