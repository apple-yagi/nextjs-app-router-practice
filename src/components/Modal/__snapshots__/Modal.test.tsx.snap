// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > Modalが正しく表示されること 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="fixed left-1/2 top-1/2 w-[calc(100vw_-_32px)] max-w-[480px] -translate-x-1/2 -translate-y-1/2 transform rounded-lg bg-white p-4 shadow-lg"
  data-state="open"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <h2
    class="mb-2 text-lg font-semibold"
    id="radix-:r1:"
  >
    Modal Title
  </h2>
  <button
    aria-label="閉じる"
    class="absolute right-4 top-4 rounded-full"
    type="button"
  >
    <svg
      fill="none"
      height="15"
      viewBox="0 0 15 15"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
        fill="#6b7280"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
